pkgs/tools/networking/wireguard-go/default.nix

```diff
@@ -2,17 +2,15 @@
 
 buildGoPackage rec {
```

Comment:
Can we use buildGoModule?

Reply:
Yes I dislike this FOD-approach as well, but since the code generator starts causing issues, let's be pragmatic here and use `buildGoModule`. Also, what's the PR's title supposed to tell?

Reply:
I meant to say that package was previously done with `buildGoPackage` and for that previous version `buildGoModule` also works.

But for newer version, I was only able make `buildGoPackage` to build correctly (after adding additional dependencies from `/tun/netstack/` subdirectory, that has it's own "go.mod" and failed the build with "imports not found"

And `buildGoModule` for newer version doesn't seem to work, as I tried to integrate it after the initial comment

(and I'll modify title to make it informative)

Reply:
Please keep in mind that vgo2nix is unmaintained and does not work for every package.